src/models/messagemodel.cpp

```diff
@@ -327,7 +328,7 @@ void MessageModel::removeChannelMessage(const ChannelID &channelId, const Messag
     }
 }
 
-void MessageModel::setThreadMessages(const ChannelID &channelId, const ThreadID &threadId, const QVariantList &messages)
+void MessageModel::setThreadMessages(const ChannelID &channelId, const ThreadID &threadId, QVariantList &messages)
```

Comment:
You removed `const`, so modifying the `messages` list will modify it on the caller side as well, which is certainly unwanted. Either keep this a const ref and create a mutable copy inside the function, or remove the reference completely and pass `messages` by value (I'd prefer the first option, though).

Reply:
Yes, I need to figure out if that is indeed needed (e.g. there was a one-off index assert triggered before changing that const).
I'll get back with more details & update.

Reply:
I've re-worked this by excluding the thread-leader from the caller, as the messagemodel.h already said:)